records_mover/records/types.py

```diff
@@ -11,92 +14,89 @@
 
-    from typing_extensions import Literal  # noqa
 
-    from mypy_extensions import TypedDict
+class UrlDetailsEntry(TypedDict):
+    content_length: int
 
-    #
-    # Note: Any expansion of these types should also be done in
-    # records.jobs.hints
-    #
-    RecordsFormatType = Literal['delimited', 'parquet']
 
-    DelimitedVariant = Literal['dumb', 'csv', 'bigquery', 'bluelabs', 'vertica']
+Url = str
+UrlDetails = Dict[Url, UrlDetailsEntry]
 
-    HintEncoding = Literal["UTF8", "UTF16", "UTF16LE", "UTF16BE",
-                           "UTF16BOM", "UTF8BOM", "LATIN1", "CP1252"]
 
-    HintQuoting = Literal["all", "minimal", "nonnumeric", None]
+#
+# Note: Any expansion of these types should also be done in
+# records.jobs.hints
+#
+RecordsFormatType = Literal['delimited', 'parquet']
 
-    HintEscape = Literal["\\", None]
+VALID_VARIANTS = ['dumb', 'csv', 'bigquery', 'bluelabs', 'vertica']
+DelimitedVariant = Literal['dumb', 'csv', 'bigquery', 'bluelabs', 'vertica']
 
-    HintCompression = Literal['GZIP', 'BZIP', 'LZO', None]
+HintEncoding = Literal["UTF8", "UTF16", "UTF16LE", "UTF16BE",
+                       "UTF16BOM", "UTF8BOM", "LATIN1", "CP1252"]
 
-    # The trick here works on Literal[True, False] but not on bool:
-    #
-    # https://github.com/python/mypy/issues/6366#issuecomment-560369716
-    HintHeaderRow = Literal[True, False]
+HintQuoting = Literal["all", "minimal", "nonnumeric", None]
 
-    HintDoublequote = Literal[True, False]
+HintEscape = Literal["\\", None]
 
-else:
-    RecordsManifestEntryMetadata = Mapping[str, int]
-    LegacyRecordsManifestEntry = Mapping[str, Union[str, bool, int, RecordsManifestEntryMetadata]]
-    RecordsManifestEntryWithLength =\
-        Mapping[str, Union[str, bool, int, RecordsManifestEntryMetadata]]
-    LegacyRecordsManifest = Mapping[str, List[LegacyRecordsManifestEntry]]
-    RecordsManifestWithLength = Mapping[str, List[RecordsManifestEntryWithLength]]
-    UrlDetailsEntry = Dict[str, int]
-    Url = str
-    UrlDetails = Dict[Url, UrlDetailsEntry]
+# TODO: combine this and cli thingie
+HintCompression = Literal['GZIP', 'BZIP', 'LZO', None]
 
-    RecordsFormatType = str
+# The trick here works on Literal[True, False] but not on bool:
+#
+# https://github.com/python/mypy/issues/6366#issuecomment-560369716
+HintHeaderRow = Literal[True, False]
 
-    DelimitedVariant = str
+HintDoublequote = Literal[True, False]
 
-    HintEncoding = str
 
-    HintQuoting = Optional[str]
+# TODO: This None is a bug in the spec, right?
+HintDateFormat = Literal[None, 'YYYY-MM-DD', 'MM-DD-YYYY', 'DD-MM-YYYY', 'MM/DD/YY']
 
-    HintEscape = Optional[str]
+HintTimeOnlyFormat = Literal["HH12:MI AM", "HH24:MI:SS"]
 
-    HintCompression = Optional[str]
+HintDateTimeFormatTz = Literal["YYYY-MM-DD HH:MI:SSOF",
+                               "YYYY-MM-DD HH:MI:SS",
+                               "YYYY-MM-DD HH24:MI:SSOF",  # TODO: this is listed twice - bug in spec?
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker